File: trunk/ShadowFrameworkLite20/src/shadow/pipeline/expression/SFBasicExpressionGenerator.java
package shadow.pipeline.expression;

import java.util.List;

import shadow.pipeline.parameters.SFParameteri;


public class SFBasicExpressionGenerator implements SFExpressionGeneratori {

	@Override
	public SFExpressionOperator getOperator(String operatorSymbol){
		if(operatorSymbol.equalsIgnoreCase("+")){
			return new SFExpressionSum();
		}
		if(operatorSymbol.equalsIgnoreCase("*")){
			return new SFExpressionMult();
		}
		if(operatorSymbol.equalsIgnoreCase("/")){
			return new SFExpressionDivide();
		}
		if(operatorSymbol.equalsIgnoreCase("-")){
			return new SFExpressionMinus();
		}
		if(operatorSymbol.equalsIgnoreCase(":")){
			return new SFExpressionClamp();
		}
		if(operatorSymbol.equalsIgnoreCase("°")){
			return new SFExpressionDot();
		}
		return new SFExpressionSum();
	}

	@Override
	public SFExpressionElement getExpressionElement(String value, List<SFParameteri> set){
		return new SFExpressionVariable(value,set);
	}
	
	@Override
	public SFExpressionTypeWrapper getWrapper(short type){
		return new SFExpressionTypeWrapper(type);
	}
}

File: trunk/ShadowFrameworkJS/SF20Core/src/shadow/pipeline/expression/SFBasicExpressionGenerator.js

function SFBasicExpressionGenerator(){
}

SFBasicExpressionGenerator.prototype = {

	getOperator:function(operatorSymbol){
		if(operatorSymbol.equalsIgnoreCase("+")){
		return ,new ,SFExpressionSum();
	}
		if(operatorSymbol.equalsIgnoreCase("*")){
		return ,new ,SFExpressionMult();
	}
		if(operatorSymbol.equalsIgnoreCase("/")){
		return ,new ,SFExpressionDivide();
	}
		if(operatorSymbol.equalsIgnoreCase("-")){
		return ,new ,SFExpressionMinus();
	}
	//if(operatorSymbol.equalsIgnoreCase(":"));//Warning: Not well Identified 
		return ,new ,SFExpressionClamp();
	//}
	//if(operatorSymbol.equalsIgnoreCase("°"));//Warning: Not well Identified 
		return ,new ,SFExpressionDot();
	//}
		if(operatorSymbol.equalsIgnoreCase(",")){
		return ,new ,SFExpressionVector();
	}
		return ,new ,SFExpressionSum();
	},

	getExpressionElement:function(value, set){
		return ,new ,SFExpressionVariable(value,set);
	},

	getWrapper:function(type){
		return ,new ,SFExpressionTypeWrapper(type);
	}

};